gateway/webhook/webhook_test.go

```diff
@@ -0,0 +1,228 @@
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/gateway/response"
+	mock_webhook "github.com/rudderlabs/rudder-server/mocks/gateway/webhook"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	sampleWriteKey = "SampleWriteKey"
+	sourceDefName  = "webhook"
+	sampleError    = "someError"
+	sampleOutput   = "{\"output\":true}"
+)
+
+var (
+	whStats           *webhookStatsT
+	once              sync.Once
+	transformerServer = httptest.NewServer(http.HandlerFunc(transformMockHandler))
+)
+
+func initWebhook() {
+	once.Do(func() {
+		config.Load()
+		logger.Init()
+		misc.Init()
+		Init()
+		stats.DefaultStats = &stats.HandleT{}
+		whStats = newWebhookStats()
+		sourceTransformerURL = transformerServer.URL
+	})
+}
+
+func createWebhookHandler(gwHandle GatewayI) *HandleT {
+	return &HandleT{
+		requestQ:      map[string]chan *webhookT{},
+		batchRequestQ: make(chan *batchWebhookT),
+		netClient:     retryablehttp.NewClient(),
+		gwHandle:      gwHandle,
+	}
+}
+
+func transformMockHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	reqBody, _ := io.ReadAll(r.Body)
+	_, err := w.Write(reqBody)
+	if err != nil {
+		pkgLogger.Error(err)
+	}
+}
+
+func TestWebhookRequestHandlerErrorCase(t *testing.T) {
```

Comment:
First 3 test cases don't actually use the httptest server?
Would it be possible to verify behaviour for all test cases in an end-2-end manner? e.g. given that transformer responds A, gateway responds B.